### PR TITLE
feat: m3 pro w/ bb arm benches

### PR DESCRIPTION
**Pull request by Maddiaa0**

### CODE REVIEW

File: results/m3-pro-12-arm/prove_bb_bin.csv [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+2^15,0m1.062s
+2^16,0m2.007s
+2^17,0m3.895s
+2^18,0m7.603s
+2^19,0m15.143s
+2^20,0m30.965s
+2^21,1m5.772s
+2^22,0m28.877s
+2^23,0m38.942s
+2^24,0m0.017s
```
Comment (Savio-Sou) [inline results/m3-pro-12-arm/prove_bb_bin.csv]:
```suggestion
```

Suggest deleting as barretenberg doesn't currently support downloads of trusted setups that large and these benchmarking programs aren't actually proved when running the script.

(You would receive `Failed to download g1 data.` if you attempt running the prove commands individually.)

File: results/m3-pro-12-arm/prove.csv [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+2^14,0m1.562s
+2^15,0m2.496s
+2^16,0m4.496s
+2^17,0m8.533s
+2^18,0m16.337s
+2^19,0m32.414s
+2^20,1m5.016s
+2^21,2m21.177s
+2^22,1m41.479s
+2^23,3m2.550s
```
Comment (Savio-Sou) [inline results/m3-pro-12-arm/prove.csv]:
```suggestion
```

Suggest deleting as barretenberg doesn't currently support downloads of trusted setups that large and these benchmarking programs aren't actually proved when running the script.

(You would receive `Failed to download g1 data.` if you attempt running the prove commands individually.)

**Comment by Savio-Sou** [comment]
Just to clarify, were:
- `prove.csv` and `prove_primitives.csv` collected with x86 barretenberg backend_binary (that came with `noirup -v 0.21.0`), while
- `prove_bb_bin.csv` and `prove_bb_bin_primitives.csv` collected with locally compiled arm barretenberg backend_binary?

**Comment by Maddiaa0** [comment]
> Just to clarify, were:
> 
> * `prove.csv` and `prove_primitives.csv` collected with x86 barretenberg backend_binary (that came with `noirup -v 0.21.0`), while
> * `prove_bb_bin.csv` and `prove_bb_bin_primitives.csv` collected with locally compiled arm barretenberg backend_binary?

Yes this is the case, running nargo prove does a check on the backend and downloads the one from github so its x86

**Review by Savio-Sou — APPROVED** [review]

**Pull request merged**